File: spring-boot-kafka-demo/kafka-json-demo2/src/main/java/com/github/jccode/springbootkafkademo/jsondemo2/event/Event.java
package com.github.jccode.springbootkafkademo.jsondemo2.event;

import java.io.Serializable;
import java.util.UUID;

/**
 * Event
 *
 */
public class Event<T> implements Serializable {

    /**
     * 事件ID
     */
    private String uuid;

    /**
     * 事件的名称
     */
    private String type;

    /**
     * 数据
     */
    private T payload;

    public Event() {
    }

    public Event(String uuid, String type, T payload) {
        this.uuid = uuid;
        this.type = type;
        this.payload = payload;
    }

    public Event(String type, T payload) {
        this.uuid = UUID.randomUUID().toString();
        this.type = type;
        this.payload = payload;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public T getPayload() {
        return payload;
    }

    public void setPayload(T payload) {
        this.payload = payload;
    }
}

File: spring-boot-kafka-demo/kafka-json-demo2/src/main/scala/com/github/jccode/springbootkafkademo/jsondemo2/event/Event.scala
package com.github.jccode.springbootkafkademo.jsondemo2.event

import java.util.UUID

import scala.beans.BeanProperty


class Event[T](@BeanProperty val uuid: String,
               @BeanProperty val eventType: String,
               @BeanProperty val payload: T) extends Serializable {

  // 无参构造器, java中反射用.
  def this() = this(null, null, null.asInstanceOf[T])

  def this(eventType: String, payload: T) = this(UUID.randomUUID.toString, eventType, payload)

  override def toString = s"Event($uuid, $eventType, $payload)"
}

object Event {

  def apply[T](uuid: String, eventType: String, payload: T): Event[T] = new Event(uuid, eventType, payload)

  def apply[T](eventType: String, payload: T): Event[T] = new Event(eventType, payload)

}